src/super_gradients/common/plugins/deci_client.py

```diff
@@ -71,7 +69,9 @@ def _get_file(self, model_name: str, file_name: str) -> Optional[str]:
                 logger.debug(e.body)
             return None
 
-        return FilesDataInterface.download_temporary_file(file_url=download_link)
+        file_path = FilesDataInterface.download_temporary_file(file_url=download_link)
```

Comment:
Do we want to wrap this download in wait_for_master? To ensure we do not over-download and not cause race-condition when writing to the same file?

Reply:
That's a good question.
`FilesDataInterface.download_temporary_file(...)` downloads and saves to a local temp file. This temp file should be different for every node because it is generated by each process separately. So, in the end if I'm not wrong there should be no problem race-condition wise.
It still means that we download n times every file from the platform, but they are very lightweight (it's only yamls) so I'm not sure it is worth optimizing.

Does that make sens to you?